### PR TITLE
[DSA] Add opt-in SM90 LiteTopK fused indexer top-k prefill path

### PR DESCRIPTION
**Pull request by Heisenberg-Yin**
## Motivation

This PR brings the **SM90 (Hopper) counterpart to #32094**, which added the LiteTopK fused indexer top-k prefill path for SM100. We observe a comparable speedup on H100 (1.21x–1.40x, table below).

I would also like to invite @yuan-luo to help this PR due to his help in my sm100 implementation.

The feature is **off by default**:

```bash
SGLANG_USE_SM90_LITETOPK=1
```

## Modifications

**New** — `python/sglang/srt/layers/attention/dsa/sm90_litetopk/`
- `csrc/sm90_litetopk.cuh`, `csrc/pr_hot.cuh` — the SM90 HOT12288 scorer and online bucket gate.
- `csrc/litetopk_sm90.cu`, `csrc/online_aux.cu` — launchers and auxiliary kernels.
- `__init__.py` — the Python adapter: workspace/carry-state management and the fail-closed `try_run` entry point.

**Modified**
- `dsa_indexer.py` — an `sm90_eligible` gate plus `try_sm90_litetopk()` / `observe_sm90_litetopk()` in `_get_topk_ragged`. Both the full-logits and row-chunked branches call the same adapter, so the first dense sub-chunk bootstraps subsequent HOT sub-chunks.

No packaging changes are needed: `srt/**/*` in `[tool.setuptools.package-data]` already carries the `csrc/` sources into the wheel. The CUDA sources are JIT-compiled lazily on the first eligible invocation and the adapter falls back to the dense indexer on build failure.

### Configuration

| env var | default | meaning |
|---|---|---|
| `SGLANG_USE_SM90_LITETOPK` | `0` | enable the path |
| `SM90_LITETOPK_CAPACITY` | `49152` | candidate capacity (must be >= 12288 and divisible by 256) |
| `SM90_LITETOPK_DEEP_GEMM_INCLUDE` | — | optional DeepGEMM/CUTLASS include path |
| `SM90_LITETOPK_BUILD_VERBOSE` | `0` | verbose JIT build output |

## Accuracy Tests

Validated on NVIDIA H100, PyTorch 2.8.0, CUDA 12.9:

- Real 128K capture, Q=64: **exact top-k, recall 1.0**, no invalid or duplicate indices.
- Odd Q=3 adapter path: exact top-k, recall 1.0.
- Q=17 with SGLang's native ragged selector: exact top-k, recall 1.0.
- Candidate-overflow case correctly returns to the dense path.
- Source K verified **byte-for-byte unchanged** after a rejected HOT attempt.
- Both CUDA extensions compile for `sm_90a`.

Recall is exact by construction: the bucket gate only ever tightens and stays above the bucket of the true k-th best score, so a stale threshold admits extra candidates but never drops one; the final select trims the superset exactly.

## Speed Tests and Profiling

Indexer top-k latency, NVIDIA H100 (SM90), native SGLang dense path vs. this path:

| KV length | native SGLang | SM90 LiteTopK | speedup |
|---|---|---|---|
| 128K | 11.916 ms | 9.855 ms | **1.21x** |
| 256K | 24.637 ms | 19.639 ms | **1.25x** |
| 512K | 50.696 ms | 38.018 ms | **1.33x** |
| 768K | 77.370 ms | 55.445 ms | **1.40x** |
| 1M | 103.558 ms | 74.367 ms | **1.39x** |

The speedup grows with context length, as expected — the dense path's cost is dominated by materializing and scanning the full `[num_q, seq_len]` logits matrix, which this path skips after the bootstrap chunk.

### CODE REVIEW

**Comment by Heisenberg-Yin** [comment]
@Fridge003 could you take a look when you have a chance?

This is the SM90 (Hopper) counterpart to #32094 — same LiteTopK technique, measured at 1.21x–1.40x on H100 across 128K–1M (table in the description). It is opt-in and off by default, and the adapter is fail-closed: anything outside the dispatch scope returns `None` and the existing dense path runs unchanged on the original tensors.
